enterprise/cmd/frontend/internal/repos/webhooks/bitbucket_server_handler.go

```diff
@@ -0,0 +1,77 @@
+package webhooks
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/webhooks"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/errcode"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/bitbucketserver"
+	"github.com/sourcegraph/sourcegraph/internal/repoupdater"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type BitbucketServerHandler struct {
+	logger log.Logger
+}
+
+func (g *BitbucketServerHandler) Register(router *webhooks.Router) {
+	router.Register(func(ctx context.Context, _ database.DB, _ extsvc.CodeHostBaseURL, payload any) error {
+		return g.handlePushEvent(ctx, payload)
+	}, extsvc.KindBitbucketServer, "repo:refs_changed")
+}
+
+func NewBitbucketServerHandler() *BitbucketServerHandler {
+	return &BitbucketServerHandler{
+		logger: log.Scoped("webhooks.BitbucketServerHandler", "bitbucket server webhook handler"),
+	}
+}
+
+func (g *BitbucketServerHandler) handlePushEvent(ctx context.Context, payload any) error {
+	event, ok := payload.(*bitbucketserver.PushEvent)
+	if !ok {
+		return errors.Newf("expected BitbucketServer.PushEvent, got %T", payload)
+	}
+
+	repoName, err := bitbucketServerNameFromEvent(event)
+	if err != nil {
+		return errors.Wrap(err, "handleGitLabWebhook: get name failed")
+	}
+
+	resp, err := repoupdater.DefaultClient.EnqueueRepoUpdate(ctx, repoName)
+	if err != nil {
+		// Repo not existing on Sourcegraph is fine
+		if errcode.IsNotFound(err) {
+			g.logger.Warn("BitbucketServer push webhook received for unknown repo", log.String("repo", string(repoName)))
```

Comment:
Do we want to add a hint like this for site admins looking at the logs and getting worried?
```suggestion
			g.logger.Warn("BitbucketServer push webhook received for unknown repo. Was the repo deleted?", log.String("repo", string(repoName)))
```

Reply:
I think putting this in docs might be better. IMHO this feels like it makes the logs a bit noisier.